request.go

```diff
@@ -213,6 +213,22 @@ func (r Request) ResponseWithCode(body interface{}, statusCode int) Response {
 	return rsp
 }
 
+// RouterEndpointPattern finds the router pattern that matches the request. This is only callable while the request
+// is being served.
+func (r Request) RouterEndpointPattern() string {
+	if router := RouterForRequest(r); router != nil {
+		if pathPattern := router.Pattern(r); pathPattern != "" {
+			return pathPattern
+		}
+	}
+	return ""
+}
+
+// RequestMethod returns the HTTP method of the request
```

Comment:
I'm not particularly keen on the name of this so any suggestions would be appreciated 🙇